plugin/hover.py

```diff
@@ -311,6 +313,13 @@ def _show_hover(self, listener: AbstractViewListener, point: int, only_diagnosti
                     location=point,
                     on_navigate=lambda href: self._on_navigate(href, point),
                     on_hide=lambda: self.view.erase_regions(HOVER_HIGHLIGHT_KEY))
+            self._image_resolver = mdpopups.resolve_images(
+                contents, mdpopups.worker_thread_resolver, partial(self._on_images_resolved, contents))
+
+    def _on_images_resolved(self, original_contents: str, contents: str) -> None:
+        self._image_resolver = None
+        if contents != original_contents and self.view.is_popup_visible():
+            update_lsp_popup(self.view, contents)
```

Comment:
It looks like this could trigger a situation like:
 - the popup is shown
 - images start resolving
 - the original popup is closed and a different one is opened
 - the original resolve completes and replaces the new popup with original contents.

Should we have some safety measures to ignore resolved results that are "outdated"?

Reply:
Here is a simple way to test it with Pyright:
```python
def show_cat():
    """ Cute cat photo on hover.
    
    ![](https://upload.wikimedia.org/wikipedia/commons/8/8e/Hauskatze_langhaar.jpg)
    """
    pass
```

I guess it depends on your internet connection speed, but on my end the delay is negligible, in the way that I doubt you can open another popup in the meanwhile. The image size is ~1MB.